scripts/dev/curriculum/_commands/helpers.ts

```diff
@@ -0,0 +1,238 @@
+
+import { Page } from "puppeteer";
+import sharp from "sharp";
+
+import { BASE_PATH } from "./config";
+
+export type ScreenshotModalResult = {
+  slug: string;
+  screenshot: string;
+};
+
+export type ScreenshotPageResult = {
+  slug: string;
+  screenshot: string;
+  modals: ScreenshotModalResult[];
+};
+
+export type ScreenshotResult = {
+  pages: ScreenshotPageResult[];
+  includesModals: boolean;
+};
+
+export function wrapHtmlLayout(content: string) {
+  return `
+      <html>
+        <body>
+          <style>
+              html {
+                box-sizing: border-box;
+                font-family: sans-serif;
+                font-size: 14px;
+              }
+              html, body {
+                margin: 0;
+                padding: 0;
+              }
+              *, *:before, *:after {
+                box-sizing: inherit;
+              }
+              .column-header{
+                background: #eeeeee;
+                padding: 8px;
+                position: sticky;
+                top: 8px;
+                z-index: 9999;
+                border-radius: 5px;
+              }
+              .nav-item{
+                margin:8px 0;
+                display:flex;
+                align-items: center;
+              }
+              .nav-item a {
+                text-wrap: nowrap;
+                color:#333;
+              }
+              .nav-item a:hover{
+                color: #777;
+              }
+              .pill{
+                background:#333;
+                color:#fff;
+                padding:2px 4px;
+                border-radius:3px;
+              }
+              .image-holder{
+                position: relative;
+                overflow: hidden;
+              }
+              .diff-image{
+                position: absolute;
+                top: 0;
+                left: 0;
+                bottom: 0;
+                right: 0;
+                mix-blend-mode: difference;
+              }
+              .image-holder img{
+                width: 100%;
+              }
+          </style>
+          <div style="display: flex;">
+            ${content}
+          </div>
+        </body>
+      </html>
+    `;
+}
+
+export function resolveInputPath(input: string) {
+  if (input.startsWith(":")) {
+    const labelName = input.replace(/^:/, "");
+    return `${BASE_PATH}/screenshots/${labelName}/`;
+  } else {
+    return input;
+  }
+}
+
+export async function readJson(filepath: string) {
+  return JSON.parse((await readFile(filepath)).toString());
+}
+
+export function removeFileExtension(filename: string) {
+  return filename.substring(0, filename.lastIndexOf(".")) || filename;
+}
+
+type Comparison = {
+  slug: string;
+  screenshot: string;
+};
+export function renderComparison(
+  baseLabel: string,
+  targetLabel: string,
+  before: Comparison | null,
+  after: Comparison | null,
+) {
+  if (!before || !after) {
+    throw new Error("Missing");
```

Comment:
As elsewhere, we could add a more descriptive error message here

Reply:
Fixed, thanks!